subcommand/mesh-init/command_test.go

```diff
@@ -972,6 +1009,24 @@ func constructContainerResponse(name, health string) awsutil.ECSTaskMetaContaine
 	}
 }
 
+// stopDataplaneContainer marks the dataplane container's status as STOPPED in the
+// task meta response
+func stopDataplaneContainer(taskMetadataResp *awsutil.ECSTaskMeta) {
+	index := -1
+	for i, c := range taskMetadataResp.Containers {
+		if isDataplaneContainer(c) {
+			index = i
+			break
```

Comment:
nit: simplification - do the work in the array and return (so we don't need to track a found index)

```suggestion
			taskMetadataResp.Containers[i].DesiredStatus = ecs.DesiredStatusStopped
			taskMetadataResp.Containers[i].KnownStatus = ecs.DesiredStatusStopped
			return
```

And if `c` is not a copy, then this could just be `c.DesiredStatus = ...` instead of reindexing the array.

Another option is [`slices.IndexFunc`](https://pkg.go.dev/golang.org/x/exp/slices#IndexFunc) (note: experimental package but fine for tests) where I think this loop could be replaced by `index = slices.IndexFunc(taskMetadataResp.Containers, isDataplaneContainer)`